Client/Frontend/Toolbar+URLBar/OverlayModeManager.swift

```diff
@@ -4,20 +4,15 @@
 
 import Foundation
 
-protocol OverlayModeManager {
+protocol OverlayStateProtocol {
     var inOverlayMode: Bool { get }
-    func enterOverlayMode(_ locationText: String?, pasted: Bool, search: Bool)
-    func leaveOverlayMode(didCancel cancel: Bool)
 }
 
-extension OverlayModeManager {
-    func enterOverlayMode(_ locationText: String? = nil, pasted: Bool = false, search: Bool = true) {
-        enterOverlayMode(locationText, pasted: pasted, search: search)
-    }
-
-    func leaveOverlayMode(didCancel cancel: Bool = false) {
-        leaveOverlayMode(didCancel: cancel)
-    }
+protocol OverlayModeManager: OverlayStateProtocol {
+    func pasteContent(pasteContent: String)
+    func openNewTab(_ locationText: String?, url: URL?)
+    func finishEdition(didCancel: Bool)
+    func switchTab(didCancel: Bool)
 }
```

Comment:
Swift documentation on those pls 🙏 Will help us know when we should use some functions over another one